airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/ssh/SshTunnel.java

```diff
@@ -368,16 +368,21 @@ ClientSession openTunnel(final SshClient client) {
           remoteServiceHost, remoteServicePort, address.toInetSocketAddress()));
       return session;
     } catch (final IOException | GeneralSecurityException e) {
-      if (e instanceof SshException && e.getMessage()
-          .toLowerCase(Locale.ROOT)
-          .contains("failed to get operation result within specified timeout")) {
+      if (e instanceof SshException &&
+          (isTimeout(e) || e.getMessage().contains("Failed (ConnectException) to execute: Connection refused"))) {
```

Comment:
Since both of these conditions result in a timeout config error exception, I think it makes sense to wrap all the logic into your `isTimeout` method